### ISSUE TITLE
ICE: Failed to normalize Alias with type error during match checking (invalid TAIT usage)

### ISSUE THREAD

**Issue by trait-Cong** [rust-lang/rust#150770]
I tried this code:

```rust

#![feature(type_alias_impl_trait)]
trait Trait<T> {}
type Alias<'a, U: Trait> = impl Trait<U>;
pub enum UninhabitedVariants {
    Tuple(Alias),
}
#[define_opaque(Alias)]
fn uwu(x: UninhabitedVariants) {
    match x {}
}
fn main() {}

```

I expected to see this happen: The compiler should report the errors

Instead, this happened:The compiler emits the errors but then panics with an ICE.

### Meta

`rustc --version --verbose`:
```
rustc 1.92.0 (ded5c06cf 2025-12-08)
binary: rustc
commit-hash: ded5c06cf21d2b93bffd5d884aa6e96934ee4234
commit-date: 2025-12-08
host: x86_64-unknown-linux-gnu
release: 1.92.0
LLVM version: 21.1.3

```

<details><summary>Backtrace</summary>
<p>

```
error[E0658]: use of unstable library feature `type_alias_impl_trait`: `type_alias_impl_trait` has open design concerns
 --> Traitor/trait-fuzzer/results/crash/case_iter_88_var_4/after.rs:6:3
  |
6 | #[define_opaque(Alias)]
  |   ^^^^^^^^^^^^^
  |
  = note: see issue #63063 <https://github.com/rust-lang/rust/issues/63063> for more information

error[E0106]: missing lifetime specifier
 --> Traitor/trait-fuzzer/results/crash/case_iter_88_var_4/after.rs:4:11
  |
4 |     Tuple(Alias),
  |           ^^^^^ expected named lifetime parameter
  |
help: consider introducing a named lifetime parameter
  |
3 ~ pub enum UninhabitedVariants<'a> {
4 ~     Tuple(Alias<'a>),
  |

error[E0658]: `impl Trait` in type aliases is unstable
 --> Traitor/trait-fuzzer/results/crash/case_iter_88_var_4/after.rs:2:28
  |
2 | type Alias<'a, U: Trait> = impl Trait<U>;
  |                            ^^^^^^^^^^^^^
  |
  = note: see issue #63063 <https://github.com/rust-lang/rust/issues/63063> for more information

error[E0107]: missing generics for trait `Trait`
 --> Traitor/trait-fuzzer/results/crash/case_iter_88_var_4/after.rs:2:19
  |
2 | type Alias<'a, U: Trait> = impl Trait<U>;
  |                   ^^^^^ expected 1 generic argument
  |
note: trait defined here, with 1 generic parameter: `T`
 --> Traitor/trait-fuzzer/results/crash/case_iter_88_var_4/after.rs:1:7
  |
1 | trait Trait<T> {}
  |       ^^^^^ -
help: add missing generic argument
  |
2 | type Alias<'a, U: Trait<T>> = impl Trait<U>;
  |                        +++

error[E0107]: missing generics for type alias `Alias`
 --> Traitor/trait-fuzzer/results/crash/case_iter_88_var_4/after.rs:4:11
  |
4 |     Tuple(Alias),
  |           ^^^^^ expected 1 generic argument
  |
note: type alias defined here, with 1 generic parameter: `U`
 --> Traitor/trait-fuzzer/results/crash/case_iter_88_var_4/after.rs:2:6
  |
2 | type Alias<'a, U: Trait> = impl Trait<U>;
  |      ^^^^^     -
help: add missing generic argument
  |
4 |     Tuple(Alias<U>),
  |                +++

error: internal compiler error: compiler/rustc_middle/src/ty/normalize_erasing_regions.rs:171:13: Failed to normalize Alias(Free, AliasTy { args: ['{erased}, {type error}], def_id: DefId(0:5 ~ after[4c23]::Alias), .. }) in typing_env=TypingEnv { typing_mode: Analysis { defining_opaque_types_and_generators: [] }, param_env: ParamEnv { caller_bounds: [] } }, maybe try to call `try_normalize_erasing_regions` instead


thread 'rustc' (279199) panicked at compiler/rustc_middle/src/ty/normalize_erasing_regions.rs:171:13:
Box<dyn Any>
stack backtrace:
   0:     0x7f6f9f033ab3 - <std::sys::backtrace::BacktraceLock::print::DisplayBacktrace as core::fmt::Display>::fmt::h718e2d17a1928e63
   1:     0x7f6f9f801b98 - core::fmt::write::h1d2246b072ea91eb
   2:     0x7f6fa0dcd351 - std::io::Write::write_fmt::haf55272405c09d9b
   3:     0x7f6f9eff9642 - std::sys::backtrace::BacktraceLock::print::h61c3bd81a9458a03
   4:     0x7f6f9efff709 - std::panicking::default_hook::{{closure}}::haf1ffb5d1e33a97f
   5:     0x7f6f9efff233 - std::panicking::default_hook::hc32245deb6eaa988
   6:     0x7f6f9dffd651 - std[836535f787e97d3]::panicking::update_hook::<alloc[bc4a0c534c311190]::boxed::Box<rustc_driver_impl[88352806430fedcc]::install_ice_hook::{closure#1}>>::{closure#0}
   7:     0x7f6f9efffb2f - std::panicking::panic_with_hook::h43adc00fd0e494cb
   8:     0x7f6f9e037211 - std[836535f787e97d3]::panicking::begin_panic::<rustc_errors[bfaf558bc9da2898]::ExplicitBug>::{closure#0}
   9:     0x7f6f9e02b296 - std[836535f787e97d3]::sys::backtrace::__rust_end_short_backtrace::<std[836535f787e97d3]::panicking::begin_panic<rustc_errors[bfaf558bc9da2898]::ExplicitBug>::{closure#0}, !>
  10:     0x7f6f9e02b097 - std[836535f787e97d3]::panicking::begin_panic::<rustc_errors[bfaf558bc9da2898]::ExplicitBug>
  11:     0x7f6f9e042b01 - <rustc_errors[bfaf558bc9da2898]::diagnostic::BugAbort as rustc_errors[bfaf558bc9da2898]::diagnostic::EmissionGuarantee>::emit_producing_guarantee
  12:     0x7f6f9e5de71a - rustc_middle[cc5e45d0ca739427]::util::bug::opt_span_bug_fmt::<rustc_span[4aa1f42928653b61]::span_encoding::Span>::{closure#0}
  13:     0x7f6f9e5de9fa - rustc_middle[cc5e45d0ca739427]::ty::context::tls::with_opt::<rustc_middle[cc5e45d0ca739427]::util::bug::opt_span_bug_fmt<rustc_span[4aa1f42928653b61]::span_encoding::Span>::{closure#0}, !>::{closure#0}
  14:     0x7f6f9e5cf9bb - rustc_middle[cc5e45d0ca739427]::ty::context::tls::with_context_opt::<rustc_middle[cc5e45d0ca739427]::ty::context::tls::with_opt<rustc_middle[cc5e45d0ca739427]::util::bug::opt_span_bug_fmt<rustc_span[4aa1f42928653b61]::span_encoding::Span>::{closure#0}, !>::{closure#0}, !>
  15:     0x7f6f9c3811c0 - rustc_middle[cc5e45d0ca739427]::util::bug::bug_fmt
  16:     0x7f6f9f84e7c9 - <rustc_middle[cc5e45d0ca739427]::ty::normalize_erasing_regions::NormalizeAfterErasingRegionsFolder as rustc_type_ir[5960852b3f357f42]::fold::TypeFolder<rustc_middle[cc5e45d0ca739427]::ty::context::TyCtxt>>::fold_ty
  17:     0x7f6f9fd05091 - <rustc_pattern_analysis[98fdd7650790dbb9]::rustc::RustcPatCtxt as rustc_pattern_analysis[98fdd7650790dbb9]::PatCx>::ctor_sub_tys
  18:     0x7f6f9fd05443 - <rustc_pattern_analysis[98fdd7650790dbb9]::pat::WitnessPat<rustc_pattern_analysis[98fdd7650790dbb9]::rustc::RustcPatCtxt>>::wild_from_ctor
  19:     0x7f6f9fd11c4f - <rustc_pattern_analysis[98fdd7650790dbb9]::usefulness::WitnessMatrix<rustc_pattern_analysis[98fdd7650790dbb9]::rustc::RustcPatCtxt>>::apply_constructor
  20:     0x7f6f9fd15de3 - rustc_pattern_analysis[98fdd7650790dbb9]::usefulness::compute_exhaustiveness_and_usefulness::<rustc_pattern_analysis[98fdd7650790dbb9]::rustc::RustcPatCtxt>::{closure#0}
  21:     0x7f6f9fd0ff80 - rustc_pattern_analysis[98fdd7650790dbb9]::rustc::analyze_match
  22:     0x7f6f9c33f7c5 - <rustc_mir_build[9780538c53e38088]::thir::pattern::check_match::MatchVisitor>::check_match
  23:     0x7f6f9fd06669 - <rustc_mir_build[9780538c53e38088]::thir::pattern::check_match::MatchVisitor as rustc_middle[cc5e45d0ca739427]::thir::visit::Visitor>::visit_expr
  24:     0x7f6f9fd06710 - <rustc_mir_build[9780538c53e38088]::thir::pattern::check_match::MatchVisitor as rustc_middle[cc5e45d0ca739427]::thir::visit::Visitor>::visit_expr
  25:     0x7f6f9fd0663f - <rustc_mir_build[9780538c53e38088]::thir::pattern::check_match::MatchVisitor as rustc_middle[cc5e45d0ca739427]::thir::visit::Visitor>::visit_expr
  26:     0x7f6f9fd06710 - <rustc_mir_build[9780538c53e38088]::thir::pattern::check_match::MatchVisitor as rustc_middle[cc5e45d0ca739427]::thir::visit::Visitor>::visit_expr
  27:     0x7f6f9fd0663f - <rustc_mir_build[9780538c53e38088]::thir::pattern::check_match::MatchVisitor as rustc_middle[cc5e45d0ca739427]::thir::visit::Visitor>::visit_expr
  28:     0x7f6f9fd0888b - rustc_mir_build[9780538c53e38088]::thir::pattern::check_match::check_match
  29:     0x7f6f9fd0851b - rustc_query_impl[f5b648d03701713a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[f5b648d03701713a]::query_impl::check_match::dynamic_query::{closure#2}::{closure#0}, rustc_middle[cc5e45d0ca739427]::query::erase::Erased<[u8; 1usize]>>
  30:     0x7f6fa003b5af - rustc_query_system[701c55d052347103]::query::plumbing::try_execute_query::<rustc_query_impl[f5b648d03701713a]::DynamicConfig<rustc_data_structures[99519894e5253d88]::vec_cache::VecCache<rustc_span[4aa1f42928653b61]::def_id::LocalDefId, rustc_middle[cc5e45d0ca739427]::query::erase::Erased<[u8; 1usize]>, rustc_query_system[701c55d052347103]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[f5b648d03701713a]::plumbing::QueryCtxt, false>
  31:     0x7f6fa003b27d - rustc_query_impl[f5b648d03701713a]::query_impl::check_match::get_query_non_incr::__rust_end_short_backtrace
  32:     0x7f6f9fb870a6 - rustc_mir_build[9780538c53e38088]::builder::build_mir
  33:     0x7f6f9f8073c1 - rustc_query_impl[f5b648d03701713a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[f5b648d03701713a]::query_impl::mir_built::dynamic_query::{closure#2}::{closure#0}, rustc_middle[cc5e45d0ca739427]::query::erase::Erased<[u8; 8usize]>>
  34:     0x7f6fa02bc5b8 - rustc_query_system[701c55d052347103]::query::plumbing::try_execute_query::<rustc_query_impl[f5b648d03701713a]::DynamicConfig<rustc_data_structures[99519894e5253d88]::vec_cache::VecCache<rustc_span[4aa1f42928653b61]::def_id::LocalDefId, rustc_middle[cc5e45d0ca739427]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[701c55d052347103]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[f5b648d03701713a]::plumbing::QueryCtxt, false>
  35:     0x7f6fa02bc009 - rustc_query_impl[f5b648d03701713a]::query_impl::mir_built::get_query_non_incr::__rust_end_short_backtrace
  36:     0x7f6f9f832972 - rustc_mir_transform[edc82e38c1b94685]::ffi_unwind_calls::has_ffi_unwind_calls
  37:     0x7f6f9f83235f - rustc_query_impl[f5b648d03701713a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[f5b648d03701713a]::query_impl::has_ffi_unwind_calls::dynamic_query::{closure#2}::{closure#0}, rustc_middle[cc5e45d0ca739427]::query::erase::Erased<[u8; 1usize]>>
  38:     0x7f6fa003b5af - rustc_query_system[701c55d052347103]::query::plumbing::try_execute_query::<rustc_query_impl[f5b648d03701713a]::DynamicConfig<rustc_data_structures[99519894e5253d88]::vec_cache::VecCache<rustc_span[4aa1f42928653b61]::def_id::LocalDefId, rustc_middle[cc5e45d0ca739427]::query::erase::Erased<[u8; 1usize]>, rustc_query_system[701c55d052347103]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[f5b648d03701713a]::plumbing::QueryCtxt, false>
  39:     0x7f6fa003b13d - rustc_query_impl[f5b648d03701713a]::query_impl::has_ffi_unwind_calls::get_query_non_incr::__rust_end_short_backtrace
  40:     0x7f6f9cb68247 - rustc_mir_transform[edc82e38c1b94685]::mir_promoted
  41:     0x7f6fa0206592 - rustc_query_impl[f5b648d03701713a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[f5b648d03701713a]::query_impl::mir_promoted::dynamic_query::{closure#2}::{closure#0}, rustc_middle[cc5e45d0ca739427]::query::erase::Erased<[u8; 16usize]>>
  42:     0x7f6fa020683e - rustc_query_system[701c55d052347103]::query::plumbing::try_execute_query::<rustc_query_impl[f5b648d03701713a]::DynamicConfig<rustc_data_structures[99519894e5253d88]::vec_cache::VecCache<rustc_span[4aa1f42928653b61]::def_id::LocalDefId, rustc_middle[cc5e45d0ca739427]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[701c55d052347103]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[f5b648d03701713a]::plumbing::QueryCtxt, false>
  43:     0x7f6fa02063e2 - rustc_query_impl[f5b648d03701713a]::query_impl::mir_promoted::get_query_non_incr::__rust_end_short_backtrace
  44:     0x7f6fa095d763 - rustc_borrowck[d1b50afb7fb5e9b2]::mir_borrowck
  45:     0x7f6fa095d619 - rustc_query_impl[f5b648d03701713a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[f5b648d03701713a]::query_impl::mir_borrowck::dynamic_query::{closure#2}::{closure#0}, rustc_middle[cc5e45d0ca739427]::query::erase::Erased<[u8; 8usize]>>
  46:     0x7f6fa02bc5b8 - rustc_query_system[701c55d052347103]::query::plumbing::try_execute_query::<rustc_query_impl[f5b648d03701713a]::DynamicConfig<rustc_data_structures[99519894e5253d88]::vec_cache::VecCache<rustc_span[4aa1f42928653b61]::def_id::LocalDefId, rustc_middle[cc5e45d0ca739427]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[701c55d052347103]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[f5b648d03701713a]::plumbing::QueryCtxt, false>
  47:     0x7f6fa02ba6c9 - rustc_query_impl[f5b648d03701713a]::query_impl::mir_borrowck::get_query_non_incr::__rust_end_short_backtrace
  48:     0x7f6f9e178244 - <rustc_hir_analysis[9991eadac4024967]::collect::type_of::opaque::TaitConstraintLocator>::check
  49:     0x7f6f9e1b8570 - rustc_hir_analysis[9991eadac4024967]::collect::type_of::opaque::find_opaque_ty_constraints_for_tait
  50:     0x7f6fa17f595d - rustc_hir_analysis[9991eadac4024967]::collect::type_of::type_of_opaque.cold
  51:     0x7f6fa0aca309 - rustc_query_impl[f5b648d03701713a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[f5b648d03701713a]::query_impl::type_of_opaque::dynamic_query::{closure#2}::{closure#0}, rustc_middle[cc5e45d0ca739427]::query::erase::Erased<[u8; 8usize]>>
  52:     0x7f6f9f81b346 - rustc_query_system[701c55d052347103]::query::plumbing::try_execute_query::<rustc_query_impl[f5b648d03701713a]::DynamicConfig<rustc_query_system[701c55d052347103]::query::caches::DefIdCache<rustc_middle[cc5e45d0ca739427]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[f5b648d03701713a]::plumbing::QueryCtxt, false>
  53:     0x7f6fa09b06db - rustc_query_impl[f5b648d03701713a]::query_impl::type_of_opaque::get_query_non_incr::__rust_end_short_backtrace
  54:     0x7f6fa0082b77 - rustc_hir_analysis[9991eadac4024967]::collect::type_of::type_of
  55:     0x7f6f9f81c6a6 - rustc_query_impl[f5b648d03701713a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[f5b648d03701713a]::query_impl::type_of::dynamic_query::{closure#2}::{closure#0}, rustc_middle[cc5e45d0ca739427]::query::erase::Erased<[u8; 8usize]>>
  56:     0x7f6f9f81b346 - rustc_query_system[701c55d052347103]::query::plumbing::try_execute_query::<rustc_query_impl[f5b648d03701713a]::DynamicConfig<rustc_query_system[701c55d052347103]::query::caches::DefIdCache<rustc_middle[cc5e45d0ca739427]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[f5b648d03701713a]::plumbing::QueryCtxt, false>
  57:     0x7f6f9f81aeef - rustc_query_impl[f5b648d03701713a]::query_impl::type_of::get_query_non_incr::__rust_end_short_backtrace
  58:     0x7f6fa0acad6c - rustc_hir_analysis[9991eadac4024967]::check::check::check_opaque
  59:     0x7f6fa0044b98 - rustc_hir_analysis[9991eadac4024967]::check::check::check_item_type
  60:     0x7f6fa003bfba - rustc_hir_analysis[9991eadac4024967]::check::wfcheck::check_well_formed
  61:     0x7f6fa003bf9b - rustc_query_impl[f5b648d03701713a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[f5b648d03701713a]::query_impl::check_well_formed::dynamic_query::{closure#2}::{closure#0}, rustc_middle[cc5e45d0ca739427]::query::erase::Erased<[u8; 1usize]>>
  62:     0x7f6fa003b812 - rustc_query_system[701c55d052347103]::query::plumbing::try_execute_query::<rustc_query_impl[f5b648d03701713a]::DynamicConfig<rustc_data_structures[99519894e5253d88]::vec_cache::VecCache<rustc_span[4aa1f42928653b61]::def_id::LocalDefId, rustc_middle[cc5e45d0ca739427]::query::erase::Erased<[u8; 1usize]>, rustc_query_system[701c55d052347103]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[f5b648d03701713a]::plumbing::QueryCtxt, false>
  63:     0x7f6fa003b316 - rustc_query_impl[f5b648d03701713a]::query_impl::check_well_formed::get_query_non_incr::__rust_end_short_backtrace
  64:     0x7f6fa0038a50 - rustc_hir_analysis[9991eadac4024967]::check::wfcheck::check_type_wf
  65:     0x7f6fa00388fd - rustc_query_impl[f5b648d03701713a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[f5b648d03701713a]::query_impl::check_type_wf::dynamic_query::{closure#2}::{closure#0}, rustc_middle[cc5e45d0ca739427]::query::erase::Erased<[u8; 1usize]>>
  66:     0x7f6fa089f794 - rustc_query_system[701c55d052347103]::query::plumbing::try_execute_query::<rustc_query_impl[f5b648d03701713a]::DynamicConfig<rustc_query_system[701c55d052347103]::query::caches::SingleCache<rustc_middle[cc5e45d0ca739427]::query::erase::Erased<[u8; 1usize]>>, false, false, false>, rustc_query_impl[f5b648d03701713a]::plumbing::QueryCtxt, false>
  67:     0x7f6fa089f576 - rustc_query_impl[f5b648d03701713a]::query_impl::check_type_wf::get_query_non_incr::__rust_end_short_backtrace
  68:     0x7f6fa03b2c3a - rustc_hir_analysis[9991eadac4024967]::check_crate
  69:     0x7f6fa02b5bb0 - rustc_interface[3dfce13d5179a6f8]::passes::analysis
  70:     0x7f6fa02b5877 - rustc_query_impl[f5b648d03701713a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[f5b648d03701713a]::query_impl::analysis::dynamic_query::{closure#2}::{closure#0}, rustc_middle[cc5e45d0ca739427]::query::erase::Erased<[u8; 0usize]>>
  71:     0x7f6fa08a61e0 - rustc_query_system[701c55d052347103]::query::plumbing::try_execute_query::<rustc_query_impl[f5b648d03701713a]::DynamicConfig<rustc_query_system[701c55d052347103]::query::caches::SingleCache<rustc_middle[cc5e45d0ca739427]::query::erase::Erased<[u8; 0usize]>>, false, false, false>, rustc_query_impl[f5b648d03701713a]::plumbing::QueryCtxt, false>
  72:     0x7f6fa08a5e36 - rustc_query_impl[f5b648d03701713a]::query_impl::analysis::get_query_non_incr::__rust_end_short_backtrace
  73:     0x7f6fa0b3a6ca - <rustc_interface[3dfce13d5179a6f8]::passes::create_and_enter_global_ctxt<core[2e27404414be4892]::option::Option<rustc_interface[3dfce13d5179a6f8]::queries::Linker>, rustc_driver_impl[88352806430fedcc]::run_compiler::{closure#0}::{closure#2}>::{closure#2} as core[2e27404414be4892]::ops::function::FnOnce<(&rustc_session[2fc1d960ad4ec558]::session::Session, rustc_middle[cc5e45d0ca739427]::ty::context::CurrentGcx, alloc[bc4a0c534c311190]::sync::Arc<rustc_data_structures[99519894e5253d88]::jobserver::Proxy>, &std[836535f787e97d3]::sync::once_lock::OnceLock<rustc_middle[cc5e45d0ca739427]::ty::context::GlobalCtxt>, &rustc_data_structures[99519894e5253d88]::sync::worker_local::WorkerLocal<rustc_middle[cc5e45d0ca739427]::arena::Arena>, &rustc_data_structures[99519894e5253d88]::sync::worker_local::WorkerLocal<rustc_hir[7611e5bcfcd2a5c8]::Arena>, rustc_driver_impl[88352806430fedcc]::run_compiler::{closure#0}::{closure#2})>>::call_once::{shim:vtable#0}
  74:     0x7f6fa0994be2 - rustc_interface[3dfce13d5179a6f8]::interface::run_compiler::<(), rustc_driver_impl[88352806430fedcc]::run_compiler::{closure#0}>::{closure#1}
  75:     0x7f6fa054a300 - std[836535f787e97d3]::sys::backtrace::__rust_begin_short_backtrace::<rustc_interface[3dfce13d5179a6f8]::util::run_in_thread_with_globals<rustc_interface[3dfce13d5179a6f8]::util::run_in_thread_pool_with_globals<rustc_interface[3dfce13d5179a6f8]::interface::run_compiler<(), rustc_driver_impl[88352806430fedcc]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>
  76:     0x7f6fa0549fe4 - <<std[836535f787e97d3]::thread::Builder>::spawn_unchecked_<rustc_interface[3dfce13d5179a6f8]::util::run_in_thread_with_globals<rustc_interface[3dfce13d5179a6f8]::util::run_in_thread_pool_with_globals<rustc_interface[3dfce13d5179a6f8]::interface::run_compiler<(), rustc_driver_impl[88352806430fedcc]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1} as core[2e27404414be4892]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  77:     0x7f6fa054896f - std::sys::thread::unix::Thread::new::thread_start::h4637f1bfded3ea68
  78:     0x7f6f9a094ac3 - start_thread
                               at ./nptl/pthread_create.c:442:8
  79:     0x7f6f9a1268c0 - __GI___clone3
                               at ./misc/../sysdeps/unix/sysv/linux/x86_64/clone3.S:81:0
  80:                0x0 - <unknown>

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.92.0 (ded5c06cf 2025-12-08) running on x86_64-unknown-linux-gnu

query stack during panic:
#0 [check_match] match-checking `uwu`
#1 [mir_built] building MIR for `uwu`
... and 8 other queries... use `env RUST_BACKTRACE=1` to see the full query stack
error: aborting due to 6 previous errors

Some errors have detailed explanations: E0106, E0107, E0658.
For more information about an error, try `rustc --explain E0106`.
```

</p>
</details>

**Comment by matthiaskrgr** [comment]
this was fixed in #148872
